src/Microsoft.Azure.SignalR.Common/EndpointRouters/DefaultMessageRouter.cs

```diff
@@ -53,30 +51,12 @@ public IEnumerable<ServiceEndpoint> GetEndpointsForGroup(string groupName, IEnum
         public IEnumerable<ServiceEndpoint> GetEndpointsForConnection(string connectionId, IEnumerable<ServiceEndpoint> endpoints)
         {
             // broadcast to all the endpoints and service side to do the filter
-            return endpoints;
+            return GetOnlineEndpoints(endpoints);
         }
 
-        /// <summary>
-        /// Only primary endpoints will be returned by client /negotiate
-        /// If no primary endpoint is available, promote one secondary endpoint
-        /// </summary>
-        /// <returns>The availbale endpoints</returns>
-        private ServiceEndpoint[] GetNegotiateEndpoints(IEnumerable<ServiceEndpoint> endpoints)
+        private IEnumerable<ServiceEndpoint> GetOnlineEndpoints(IEnumerable<ServiceEndpoint> endpoints)
         {
-            var primary = endpoints.Where(s => s.Online && s.EndpointType == EndpointType.Primary).ToArray();
-            if (primary.Length > 0)
-            {
-                return primary;
-            }
-
-            // All primary endpoints are offline, fallback to the first online secondary endpoint
-            var secondary = endpoints.Where(s => s.Online && s.EndpointType == EndpointType.Secondary).ToArray();
-            if (secondary.Length == 0)
-            {
-                throw new AzureSignalRNotConnectedException();
-            }
-
-            return secondary;
+            return endpoints.Where(e => e.Online);
```

Comment:
>endpoints.Where(e => e.Online) [](start = 19, length = 30)

would it be 0 and shall we throw an exception?

Reply:
How about not when it is for sending messages?

Reply:
Suggest to give customer some information so that they would be aware of sending failure is caused by no endpoint online.

Reply:
Keep the original behavior then, fix the issue later.